### PR TITLE
Add code for pipeline training with gold set utterances

### PR DESCRIPTION
**Pull request by siddhu001**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -106,14 +110,59 @@
...
-		train_intent_acc, train_intent_loss = trainer.pipeline_train_decoder(train_dataset)
-		valid_intent_acc, valid_intent_loss = trainer.pipeline_test_decoder(valid_dataset)
+		train_intent_acc, train_intent_loss = trainer.pipeline_train_decoder(train_dataset, postprocess_words,log_file="log_pipeline_postprocess.csv")
+		valid_intent_acc, valid_intent_loss = trainer.pipeline_test_decoder(valid_dataset,  postprocess_words,log_file="log_pipeline_postprocess.csv")
 
 		print("========= Results: epoch %d of %d =========" % (epoch+1, config.training_num_epochs))
 		print("*intents*| train accuracy: %.2f| train loss: %.2f| valid accuracy: %.2f| valid loss: %.2f\n" % (train_intent_acc, train_intent_loss, valid_intent_acc, valid_intent_loss) )
 
-		trainer.save_checkpoint()
+		trainer.save_checkpoint(model_path="model_state_postprocess.pth")
```
Comment (viswavi) [inline main.py:119]: Shouldn't include the "_postprocess" suffix if not using postprocess

File: main.py [hunk 2]
```diff
@@ -106,14 +110,59 @@
...
+
+	for epoch in range(config.training_num_epochs):
+		print("========= Epoch %d of %d =========" % (epoch+1, config.training_num_epochs))
+		train_intent_acc, train_intent_loss = trainer.pipeline_train_decoder(train_dataset,gold=True,log_file=log_file)
+		valid_intent_acc, valid_intent_loss = trainer.pipeline_test_decoder(valid_dataset, postprocess_words,log_file=log_file)
+
+		print("========= Results: epoch %d of %d =========" % (epoch+1, config.training_num_epochs))
+		print("*intents*| train accuracy: %.2f| train loss: %.2f| valid accuracy: %.2f| valid loss: %.2f\n" % (train_intent_acc, train_intent_loss, valid_intent_acc, valid_intent_loss) )
+		if postprocess_words:
+			trainer.save_checkpoint(model_path="only_gold_model_state_postprocess.pth")
```
Comment (viswavi) [inline main.py:149]: Could be nice to save the epoch number in the model path
```
trainer.save_checkpoint(model_path=f"only_gold_model_state_postprocess_{epoch}_epochs.pth")
```

**Review by viswavi — APPROVED** [review]

File: training.py [hunk 3]
```diff
@@ -130,11 +130,25 @@ def get_word_SLU(self, dataset, Sy_word, print_interval=100):
...
+					prev_k=0
+					for k in j:
+						if k==0:
+							continue
+						if k==prev_k:
+							continue
+						cur_list.append(k)
+						prev_k=k
+					cur_list=cur_list+([0]*(len(j)-len(cur_list)))
+					x_words_new.append(cur_list)
```
Comment (viswavi) [inline training.py:146]: Add separate function for post-process in a separate PR

**Comment by siddhu001** [comment]
Will handle comments in seperate PR.

**Pull request merged**